pkg/config/datafileprojectconfig/config.go

```diff
@@ -82,6 +82,33 @@ func (c DatafileProjectConfig) GetAttributeID(key string) string {
 	return c.attributeKeyToIDMap[key]
 }
 
+// GetAttributes returns attributes slice
+func (c DatafileProjectConfig) GetAttributes() []entities.Attribute {
+	attributes := make([]entities.Attribute, 0, len(c.attributeMap))
+	for _, attr := range c.attributeMap {
+		attributes = append(attributes, attr)
+	}
+	return attributes
+}
+
+// GetEvents returns events slice
+func (c DatafileProjectConfig) GetEvents() []entities.Event {
+	events := make([]entities.Event, 0, len(c.eventMap))
+	for _, event := range c.eventMap {
+		events = append(events, event)
+	}
+	return events
+}
+
+// GetOptimizelyAudiences returns audiences slice
+func (c DatafileProjectConfig) GetOptimizelyAudiences() []entities.OptimizelyAudience {
```

Comment:
Make sure you are combining typedAudiences with regular audiences giving precedence to the TypedAudiences. Also filtering out any audience that has the $opt_dummy_audience.

Reply:
Thanks!

Reply:
Can we move this function to OptimizelyConfig, so we can avoid depending on OptimizelyConfig's type here